### ISSUE TITLE
Incorrect documentation for JUPYTERHUB_BASE_URL

### ISSUE THREAD

**Issue by miwig** [jupyterhub/jupyterhub#4397]
### Bug description
The [documentation states](https://github.com/jupyterhub/jupyterhub/blob/a348ba6536d21752ab8f8d13d8611e14a147837f/docs/source/reference/services.md?plain=1#L126):

> JUPYTERHUB_BASE_URL:       Base URL of the Hub (https://mydomain[:port]/)

However, the variable [seems to be generated from `hub.base_url`](https://github.com/jupyterhub/jupyterhub/blob/3491ad68162cd52c198a3732a50ad6cee5a8bf2c/jupyterhub/spawner.py#L988) which does not include the domain part.

#### Expected behaviour

- The documentation should match the actual behaviour.
- It would be nice if the full public-facing URL of the hub could be passed to services as well (according to the documentation, this is what `c.JupyterHub.bind_url` is for?)

#### Actual behaviour

`JUPYTERHUB_BASE_URL` is set to the path relative to the domain, i.e. just `/`, contrary to what the documentation says.

### How to reproduce

1. Set `c.JupyterHub.bind_url`
2. Start a single-user server
3. Open a terminal and execute `echo $JUPYTERHUB_BASE_URL`

**Comment by manics** [comment]
Thanks for pointing out the incorrect docs!

> It would be nice if the full public-facing URL of the hub could be passed to services as well

JupyterHub can't automatically determine the public-facing domain since there could be proxies or load-balancers in front of it, or multiple domains could resolve to the same host.

**Comment by miwig** [comment]
> JupyterHub can't automatically determine the public-facing domain since there could be proxies or load-balancers in front of it, or multiple domains could resolve to the same host.

The documentation [states](https://jupyterhub.readthedocs.io/en/stable/api/app.html#jupyterhub.app.JupyterHub.bind_url) that `JupyterHub.bind_url` should be set to "The public facing URL of the whole JupyterHub application." If this is not correct in all cases, it should maybe also be changed?

**Comment by minrk** [comment]
Yeah, that text could be clearer - it's the _bind_ url (what's passed to `listen()`) for what will be the public entrypoint to the hub components _that jupyterhub is in control of_ (i.e. the proxy as opposed to the hub or any other internal component). Any number of proxies may be involved that JupyterHub _doesn't_ know or care about, and JupyterHub never needs to know the _connect_ url for the public entrypoint. The key bits of information are:

- whatever your public url is and however it is handled (gateways, proxies, ssl, etc.), it should ultimately arrive at this address, and not any of the others (e.g. `hub_bind_url` or `hub_connect_url`).
- it is a bind address, not a connect address (i.e. `0.0.0.0` and similar 'all-interfaces' addresses are valid and indeed default).

If anyone has a great, clear way to communicate these, I would love that!
